### PR TITLE
IGVF-2877 Group file sets in association graph logically

### PR DESCRIPTION
**Pull request by forresttanaka**
# Branch Code Review

## Model
GPT-5.3-Codex

## Scope Reviewed
Current local branch state compared to dev (including unpushed local commits).

## Findings
No actionable findings.

I re-reviewed the currently changed branch files and did not identify a concrete correctness bug or regression that should block merge.

## What Was Specifically Rechecked
- Grouping pipeline safety for stage-to-stage node map behavior.
- Related measurement grouping null-handling for missing map entries.
- Dataset summary request contract and its current tests/callers.
- Removal of the standalone fileset chart page route.

## Validation Performed
- Ran grouping pipeline tests:
  - npm run -s test -- components/file-graph/grouping-pipeline/__tests__/grouping-pipeline.test.ts
- Ran common request tests:
  - npm run -s test -- lib/__tests__/common-requests.test.ts

Both suites passed in the current branch state.

## Residual Risk / Notes
- Full workspace typecheck task in this environment still reports missing tsc in the configured fallback path, so this review relies on targeted test coverage plus static inspection rather than a clean full-repo typecheck run.

### CODE REVIEW

**Comment by forresttanaka** [comment]
# Code Review — IGVF-2877 Group file sets in association graph logically

*Reviewed by GitHub Copilot (Claude Sonnet 4.6)*

## Summary

**Approved — ready to merge.**

This PR introduces a **grouping pipeline** for the file graph that clusters related file-set nodes into visual group nodes. Measurement sets are grouped with their auxiliary sets, related measurement sets, control file sets, and construct library sets. The pipeline is implemented as a series of composable stages, each of which accepts the nodes not yet grouped by previous stages and returns the groups it found along with the remaining ungrouped nodes.

The PR also migrates `FileSetObject` out of `globals.d.ts` into `lib/file-sets.ts` with fully typed subinterfaces per file-set type (`MeasurementSetObject`, `AuxiliarySetObject`, `ConstructLibrarySetObject`, etc.), and converts several `.js` files to `.tsx`/`.ts`. No blocking issues were found.

---

## Positives

- The pipeline pattern with composable stages is well-structured and easy to extend. Adding a new grouping relationship means adding a new stage file and appending it to `groupingPipelineStages`.
- Rebuilding `nodesByFileSetPath` from `remainingNodes` at each stage (rather than all nodes) is correct and well-commented — it prevents later stages from connecting to nodes already claimed by earlier stages.
- `FileSetTypeMap` combined with `isFileSetObjectType<T>` is a clean, scalable pattern for narrowing file-set objects to their specific subtypes without needing manual `@type` array checks at every call site.
- `LinkToArray<T>` cleanly captures the API's pattern of properties that can be either an array of strings (links) or an array of embedded objects.
- Non-null assertions (`!`) are used appropriately in the DFS algorithm in `standard-grouping.ts`, making the invariants explicit: the stack only ever contains IDs pre-seeded into both `nodeById` and `adjacency`.
- Debug logging is properly gated behind `process.env.NODE_ENV === "development"`, which Next.js eliminates entirely from production bundles via dead-code elimination.
- The `fileSet.samples?.length > 0` fix in `file-set-modal.tsx` is a good defensive correction over the previous `?.length &&` form, which would have been falsy for a zero-length array but also for `undefined`.
- Test coverage for the new pipeline stages is comprehensive, including edge cases like nodes with no eligible relationships and deduplication of related measurement sets appearing in multiple arrays.

File: pages/history.tsx [hunk 1]
```diff
@@ -139,6 +140,8 @@ function Entry({
  * @param history - History object for a database object
  */
 export default function History({ history }: { history: HistoryObject }) {
+  console.log("**** HISTORY OBJECT ****", history);
+
```
[the remark this answers is not in the document]
Comment (forresttanaka) [reply pages/history.tsx]: Fixed

File: components/file-graph/grouping-pipeline/__tests__/grouping-pipeline.test.ts [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    expect(allGroupedIds).toEqual([
+      "/construct-library-sets/IGVFDS0000CLIB/",
+      "/measurement-sets/IGVFDS0000MEAS/",
+    ]);
+
+    // No remaining nodes since all nodes were grouped.
+    expect(result.remainingNodes.length).toBe(0);
+  });
+
+  it("groups a pseduobulk set with two construct library sets using their construct_library_set relationships", () => {
```
[the remark this answers is not in the document]
Comment (forresttanaka) [reply components/file-graph/grouping-pipeline/__tests__/grouping-pipeline.test.ts]: Fixed

File: lib/fetch-request.ts [hunk 3]
```diff
@@ -308,18 +312,8 @@ export default class FetchRequest {
...
-      response !== null &&
-      "@type" in response
-    ) {
-      const types = (response as DatabaseObject)["@type"];
-      return !types.includes("Error");
-    }
-    return true;
+  static isResponseSuccess<T>(response: T | ErrorObject): response is T {
+    return isDatabaseObject(response);
   }
```
[the remark this answers is not in the document]
Comment (forresttanaka) [reply lib/fetch-request.ts:324]: Fixed
